svm/src/transaction_processor.rs

```diff
@@ -440,35 +435,25 @@ impl<FG: ForkGraph> TransactionBatchProcessor<FG> {
         })?;
 
         let fee_payer_address = message.fee_payer();
-        let mut fee_payer_account = if let Some(fee_payer_account) =
-            account_overrides.and_then(|overrides| overrides.get(fee_payer_address).cloned())
-        {
-            fee_payer_account
-        } else if let Some(fee_payer_account) = callbacks.get_account_shared_data(fee_payer_address)
-        {
-            callbacks.inspect_account(
-                fee_payer_address,
-                AccountState::Alive(&fee_payer_account),
-                true, // <-- is_writable
-            );
 
-            fee_payer_account
-        } else {
+        let Some(mut loaded_fee_payer) =
+            account_loader.load_account(fee_payer_address, AccountUsagePattern::Writable)
```

Comment:
see previous question on distinction of `Writable` and `WritableInstruction`.

Does it matter that the `fee_payer_address` can also be passed to an instruction?

Reply:
as noted above, no we only need to tell `load_account` that its writable so it does the right thing with `inspect_account` and checks rent